security/tpm/test/_tpm_test.js

```diff
@@ -116,13 +202,117 @@ describe('tpm', function () {
 
     it('correctly deals with error during tpm connection', function(done) {
       /*Tests_SRS_NODE_TPM_SECURITY_CLIENT_06_013: [If `signWithIdentity` is invoked without a previous successful invocation of `activateIdentityKey`, the callback will be invoked with `err` of `InvalidOperationError`.] */
-      var client = new TpmSecurityClient('registration', fakeSimpleTpmClient);
+      var tpm = new tss.Tpm();
+      var client = new TpmSecurityClient('registration', tpm);
       client.signWithIdentity([1], (err, signResult) => {
         assert.isOk(err, 'should be an error from the connection portion of starting the tpm');
         done();
       });
     });
 
+    it('correctly deals with insufficient capabilities returned by the tpm', function(done) {
+      var fakeBadProps = {
+        tpmProperty: []
+      };
+      var tpm = new tss.Tpm();
+      var dataToSign = new Buffer('ab');
+      var connectStub = sinon.stub(tpm, 'connect');
+      connectStub.callsArgWith(0);
+      var client = new TpmSecurityClient(undefined, tpm);
+      var readPublicStub = sinon.stub(tpm, 'ReadPublic');
+      var lastResponseStub = sinon.stub(tpm, 'getLastResponseCode');
+      lastResponseStub.returns(tss.TPM_RC.SUCCESS);
+      var getPropsAndHashAlgStub = sinon.stub(client, '_getPropsAndHashAlg').callsArgWith(0,tss.TPM_ALG_ID.SHA256, fakeBadProps);
+      readPublicStub.withArgs(TpmSecurityClient._ekPersistentHandle).callsArgWith(1, fakeEkKey);
+      readPublicStub.withArgs(TpmSecurityClient._srkPersistentHandle).callsArgWith(1, fakeSrkKey);
+      readPublicStub.withArgs(TpmSecurityClient._idKeyPersistentHandle).callsArgWith(1, fakeIdKey);
+      client.signWithIdentity(dataToSign, (err, signedData) => {
+        assert.instanceOf(err, common.errors.SecurityDeviceError, 'should be an error from the signing');
```

Comment:
>'should be an error from the signing' [](start = 66, length = 37)

from the signing what? i thought it was a capability error?

Reply:
signWithIdentity

---
In reply to: [167386087](https://github.com/Azure/azure-iot-sdk-node/pull/225#discussion_r167386087) [](ancestors = 167386087)